ui/window.go

```diff
@@ -262,7 +262,16 @@ func (win *Window) drawWindowUI(gtx C) {
 // callers should not re-refresh the display.
 func (win *Window) changePage(page load.Page, keepBackStack bool) {
 	if win.currentPage != nil && keepBackStack {
-		win.currentPage.OnNavigatedFrom()
+		//When navigating to the Restore page from the MainPage, don't clear the current
+		// page (main page) data using the `currentPage.OnNavigatedFrom()` method. The
+		// Restore page will pop back to the main page when done and we'll need the data
+		// untouched. The Restore page is being opened using changePage() instead of
+		// changeFragment() because the latter renders the restore page UI on a much
+		// smaller area and distorts the display. Also, the Main page's data will NOT be reset
+		// when the Restore page is exited via popPage().
+		if page.ID() != "Restore" && win.currentPage.ID() != "Main" {
+			win.currentPage.OnNavigatedFrom()
+		}
```

Comment:
Thinking about this more, this approach looks fragile. Ideally, OnNavigatedFrom() doesn't mean the page won't be used anymore, so data required for the page to function should not be cleared in that method unless restored in the corresponding OnNavigatedTo() method.

This commit has a recommended solution: https://github.com/itswisdomagain/godcr-1/commit/a13256000d9587620fb35d7d3ae23a99bd60e1fe

Reply:
The recommendation does not work without the wallet.go file.